### ISSUE TITLE
Make requests async

### ISSUE THREAD

**Issue by reachtarunhere** [CarlQLange/chatgpt-arcana.el#7]
Hi! Thanks for the project. I love using it. 

One little issue I have is that currently whenever I make a response in the Chat my whole emacs hangs. I would ideally want to do other things nav around. I suspect this is because the requests being made are not async?

Thank You!

**Comment by CarlQLange** [comment]
Yeah I totally agree, but I was unfamiliar with the requests library when I wrote this initially. I'll try and write an async version of it tomorrow!

**Comment by CarlQLange** [comment]
I've made a messy start on this on the https://github.com/CarlQLange/chatgpt-arcana.el/tree/asyncify branch. Unfortunately I still don't really know what I'm doing, so it'll take me a bit longer to get it perfect. It's happening, though :)

**Comment by CarlQLange** [comment]
In the meantime I've seen that https://github.com/karthink/gptel is async out of the box and looks quite comparable to this package.
